From: Trond Myklebust <trond.myklebust@primarydata.com>
Date: Tue, 31 Dec 2013 13:13:30 -0500
Subject: [PATCH] SUNRPC: Ensure xprt_connect_status handles all potential
 connection errors
Git-commit: 0fe8d04e8c3a1eb49089793e38b60a17cee564e3
Patch-mainline: v3.14
References: bnc#901090

Currently, xprt_connect_status will convert connection error values such
as ECONNREFUSED, ECONNRESET, ... into EIO, which means that they never
get handled.

Signed-off-by: Trond Myklebust <trond.myklebust@primarydata.com>
Acked-by: NeilBrown <neilb@suse.de>

---
 net/sunrpc/xprt.c |    5 +++++
 1 file changed, 5 insertions(+)

--- linux-3.12-SLE12.orig/net/sunrpc/xprt.c
+++ linux-3.12-SLE12/net/sunrpc/xprt.c
@@ -755,6 +755,11 @@ static void xprt_connect_status(struct r
 	}
 
 	switch (task->tk_status) {
+	case -ECONNREFUSED:
+	case -ECONNRESET:
+	case -ECONNABORTED:
+	case -ENETUNREACH:
+	case -EHOSTUNREACH:
 	case -EAGAIN:
 		dprintk("RPC: %5u xprt_connect_status: retrying\n", task->tk_pid);
 		break;
